save-demo/src/main/kotlin/com/saveourtool/save/demo/storage/ToolStorage.kt

```diff
@@ -14,9 +17,29 @@ private const val TOOLS_PATH = "tools"
  */
 @Component
 class ToolStorage(
-    private val configProperties: ConfigProperties,
+    configProperties: ConfigProperties,
 ) : AbstractFileBasedStorage<ToolKey>(Path.of(configProperties.fileStorage.location) / TOOLS_PATH) {
-    private val rootPath = Path.of(configProperties.fileStorage.location) / TOOLS_PATH
+    /**
+     * Todo: Implement DownloadToolService
+     */
+    init {
+        val supportedTools = listOf(
+            ToolKey("diktat", "1.2.3", "diktat"),
+            ToolKey("diktat", "1.2.3", "diktat.cmd"),
+            ToolKey("ktlint", "0.47.1", "ktlint-cli"),
+            ToolKey("ktlint", "0.47.1", "ktlint-cli.cmd"),
+        )
+        list().collectList()
+            .flatMapIterable { availableFiles ->
+                supportedTools.filter {
+                    it !in availableFiles
+                }
+            }
```

Comment:
Do you essentially need to find intersection of `list()` and `supportedTools`? Then you can just `list().filter { it in supportedTools }` instead of nesting iteration

Reply:
I do not need an intersection, I need `supported \ downloaded`